### PR TITLE
ConfigSelectorMethod should check for bools

### PR DESCRIPTION
**Pull request by danielcmessias**
# resolves

https://github.com/dbt-labs/dbt-core/issues/5890 
https://github.com/dbt-labs/dbt-core/issues/5482 

### Description

The `ConfigSelectorMethod` only checks for string equality. If you have a config property that is boolean, there is no way to match it.

Consider the example:

```yaml
# schema.yml
models:
  - name: some_model
    config:
      meta:
        pii: true
```

Then running `dbt  run --select config.meta.pii:true` would not match. The only way for it work would be to encapsulate the property as a string (`pii: "true"`), not ideal.

This is a small change such that if the property is a bool, check (case-insensitively) against the strings "true" / "false".

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [ ] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

**Comment by gshank** [comment]
Could you add a test?  test/unit/test_graph_selector_parsing.py looks like a good place...

File: core/dbt/graph/selector_methods.py [hunk 1]
```diff
@@ -356,7 +356,10 @@ def search(
             except AttributeError:
                 continue
             else:
-                if selector == value:
+                # if the config is a bool then check against the strings "true"/"false"
```
Comment (jtcohen6) [inline core/dbt/graph/selector_methods.py:359]: Should we support all forms of yaml truthy/falsy (0/1, `"t`/`"f"`, `yes/no`)?

I'm fine with just `"true"` + `"false"` personally

**Review by jtcohen6 — COMMENTED** [review]
Thanks @danielcmessias! Do you feel up for adding some test cases? Relevant unit tests are [here](https://github.com/dbt-labs/dbt-core/blob/0bd6df0d1b1e4e382c413929403f7972108a5b8a/test/unit/test_graph_selector_methods.py#L771-L780).

I'm going to mark this as `ready_for_review` by Core-Execution in the meantime

File: core/dbt/graph/selector_methods.py (hunk 1 above)
Comment (danielcmessias) [reply core/dbt/graph/selector_methods.py]: I'm in favour of just true/false and avoid the yaml-specify truthy/falsy rules which don't apply in Python/JSON/Jinja templates

**Comment by danielcmessias** [comment]
@gshank @jtcohen6 Added a resolution for https://github.com/dbt-labs/dbt-core/issues/5482 and some test cases. 

Went back and forth a couple of times on the structure; idk whether separate clauses for the list and string case that both yield the node is more or less readable than one giant IF with all the conditions 🤷‍♂️ . LMK if you'd prefer it another way :)

**Comment by gshank** [comment]
Need to run pre-commit now, pip install pre-commit if it's not already installed, then run 'pre-commit run --all-files" and commit the changes.

**Review by gshank — APPROVED** [review]

**Pull request merged**